### PR TITLE
`VFS.Include` gives better hints

### PR DESCRIPTION
**Pull request by sprunk**
Says the vfs mode and interprets the "file missing" status

### CODE REVIEW

File: rts/Lua/LuaVFS.cpp [hunk 1]
```diff
@@ -150,16 +151,21 @@ int LuaVFS::Include(lua_State* L, bool synced)
 	int loadCode = 0;
 	int luaError = 0;
 
-	if ((loadCode = LoadFileWithModes(fileName, fileData, GetModes(L, 3, synced))) != 1) {
+	const auto mode = GetModes(L, 3, synced);
+	if ((loadCode = LoadFileWithModes(fileName, fileData, mode)) != 1) {
+		std::string_view hint {""};
```
Comment (badosu) [inline rts/Lua/LuaVFS.cpp:156]: Does this perform differently than:

```cpp
const std::string hint = loadCode == -1 ? "File not seen by VFS (missing or in different VFS mode)" : "";
```

**Review by badosu — APPROVED** [review]

**Comment by lhog** [comment]
Please use this opportunity and replace `SNPRINTF` with `fmt::format` or `fmt::sprintf`

File: rts/Lua/LuaVFS.cpp (hunk 1 above)
Comment (sprunk) [reply rts/Lua/LuaVFS.cpp:156]: In theory a std::string would allocate some memory and copy that string (and then pass that copy to the printf) while string_view is just a pointer to the original. A compiler would probably be able to optimize it out though.

**Comment by sprunk** [comment]
Note, `luaL_error(L, format, ...)` exists and is used elsewhere in this file, perhaps it would be possible to be consistent and use it instead of `fmt::format` (or some other way to manually build the string) + `lua_pushstring` + `lua_error(L)`. I thought about this was too scared to go that route because `VFS.Include` is somewhat special (since it executes another lua file, manipulating the stack) so maybe there is some obscure difference that would appear in some non-trivial edge case.

**Pull request merged**